src/Data/NaturalTransformation.purs

```diff
@@ -0,0 +1,14 @@
+module Data.NaturalTransformation where
+
+-- | A type for natural transformations between functors (although the `Functor`
+-- | constraint is not enforced).
+-- |
+-- | A natural transformation allows a functor to be transformed into another
```

Comment:
I'm not sure about the connection with `map`. I'd say something like "transforms a functor" and something about uniformity, and give an example like `listToMaybe`, which I think provides a decent intuition.